src/iface/route.rs

```diff
@@ -195,9 +248,7 @@ mod test {
             via_router: ADDR_1A.into(),
             preferred_until: None, expires_at: None,
         };
-        routes.update(|storage| {
-            storage.insert(cidr_1().into(), route).unwrap();
```

Comment:
So what does this achieve? One line less in route setup code? I'm not convinced it's worth adding so many functions in the public API (and you don't even validate input in some of them).

Reply:
Well, imagine you want to fetch the default gateway with the current code:
```Rust
let mut gateway = None;
routes.update(|storage| {
     let cidr = IpCidr::new(IpAddress::v4(0, 0, 0, 0), 0);
    gateway = Some(storage.get(&cidr).clone());
});
let gateway = gateway.unwrap();
```

Reply:
See the [redox netstack code after this PR](https://gitlab.redox-os.org/jD91mZM2/netstack/commit/f9d6f5d87f1f5903d301c6205b9b5637c8ad1512).

(Your existing code doesn't validate input either btw)